django/db/models/constraints.py

```diff
@@ -66,14 +72,22 @@ def deconstruct(self):
         return path, args, kwargs
 
 
+NOT_DEFERRABLE = ''
+DEFERRABLE_INITIALLY_DEFERRED = 'DEFERRABLE INITIALLY DEFERRED'
+DEFERRABLE_INITIALLY_IMMEDIATE = 'DEFERRABLE INITIALLY IMMEDIATE'
+
+
 class UniqueConstraint(BaseConstraint):
-    def __init__(self, *, fields, name, condition=None):
+    def __init__(self, *, fields, name, condition=None, defer=NOT_DEFERRABLE):
```

Comment:
What about accepting `deferrable=False` and `initially_deferred=False` kwargs instead? That seems more in line with how multi-valued options are defined in the ORM (e.g. `select_for_update`, [`bulk_create`](https://github.com/django/django/pull/9192#issuecomment-409054225)).

Reply:
It seems weird to me to use two boolean flags to model three states.

Reply:
Given `initially_deferred=True` implies `deferrable=True` both flags are unlikely to be used together.

`initially_deferred=True` translates to `DEFERRABLE INITIALLY DEFERRED`
`deferrable=True` translates to `DEFERRABLE INITIALLY IMMEDIATE`

That seems like a more consistent API with the rest of the ORM to me as we've generally avoided to expose flags pointing to constants. The only exception I can think of is `on_delete` but it's actually accepting callables and not some raw SQL defined as constants.

Reply:
I prefer the `on_delete` style api here. I take the point about not defining raw SQL.

Reply:
I still prefer the `on_delete` style, but I'm willing to change it if it's a sticking point.

Reply:
Left some comments wrt using an `Enum` which would address the concern I had above.